File: app/src/main/java/com/marceljurtz/lifecounter/views/Settings/ISettingsPresenter.java
package com.marceljurtz.lifecounter.views.Settings;

import com.marceljurtz.lifecounter.contracts.base.IPresenter;
import com.marceljurtz.lifecounter.models.Color;
import com.marceljurtz.lifecounter.enums.MagicColorEnum;

public interface ISettingsPresenter extends IPresenter {
    void onBackButtonClick();
    void onCancelButtonClick();
    void onResetButtonClick();
    void onColorSelectButtonClick(MagicColorEnum color);
    void onColorSelectValueUpdate(Color color);
    void onKeepScreenOnCheckboxClick(boolean checked);

    void onResetButtonConfirm();
    void onResetButtonCancel();

    void onMenuEntryTwoPlayerTap();
    void onMenuEntryFourPlayerTap();
    void onMenuEntryCounterManagerTap();
    void onMenuEntryDicingTap();
    void onMenuEntryAboutTap();
}

File: app/src/main/java/com/marceljurtz/lifecounter/views/Settings/ISettingsPresenter.kt
package com.marceljurtz.lifecounter.views.Settings

import com.marceljurtz.lifecounter.contracts.base.IPresenter
import com.marceljurtz.lifecounter.models.Color
import com.marceljurtz.lifecounter.enums.MagicColorEnum

interface ISettingsPresenter : IPresenter {
    fun onBackButtonClick()
    fun onCancelButtonClick()
    fun onResetButtonClick()
    fun onColorSelectButtonClick(color: MagicColorEnum)
    fun onColorSelectValueUpdate(color: Color)
    fun onKeepScreenOnCheckboxClick(checked: Boolean)
    fun onConfirmGameResetCheckboxClick(checked: Boolean)
    fun onShowAppIntroClick()

    fun onResetButtonConfirm()
    fun onResetButtonCancel()

    override fun onMenuEntryTwoPlayerTap()
    override fun onMenuEntryFourPlayerTap()
    override fun onMenuEntryCounterManagerTap()
    override fun onMenuEntryDicingTap()
    override fun onMenuEntryAboutTap()
}
